### DISCUSSION TITLE
Regarding HOPR curved meshes

### DISCUSSION THREAD

**Discussion by TRPrasanna** [loganoz/horses3d#235] [Q&A]
This is a question regarding HOPR, but I thought I would ask here anyways. Is there a straightforward way to extract the surface mesh from a CGNS file to give it as input to HOPR for the purpose of creating a higher-order curved mesh boundary? I came across this [article](https://arxiv.org/abs/2402.10913) regarding the F1 front wing simulation but it is not clear to me if the curved mesh using HOPR was generated using a surface mesh extracted from a CGNS file or if the surface mesh was generated directly from the meshing software (STAR-CCM, which I unfortunately do not have access to).

I am interested in higher-order mesh because in almost all of my airfoil simulations, my mesh around the leading-edge is a bit coarse and the Mach number erroneously reaches supersonic values around this location. For instance, see the Mach number contours around the leading-edge of a NACA4412 airfoil with the flow conditions corresponding to 14 degree angle of incidence, 1 million Reynolds number and 0.3 freestream Mach number.

![naca4412_Mach](https://github.com/user-attachments/assets/3360ecce-d726-4151-89a8-6a1d9dc12861)

I am curious to see if a curved mesh would mitigate this issue.

**Comment by loganoz** [comment]
The issue you are having seems to be related with the low order representation of the geometry you are using. You are right you can curve the mesh using hopr, but there are alternatives, like generating a high order mesh using gmsh. 
About the paper of the F1 wing, may be @Jerryntk can explain to you the combination of programs used to generate it.
